x/compliance/keeper/device_software_compliance.go

```diff
@@ -60,3 +60,19 @@ func (k Keeper) GetAllDeviceSoftwareCompliance(ctx sdk.Context) (list []types.De
 
 	return
 }
+
+func IsComplianceInfoExist(
```

Comment:
If can make it a method of `DeviceSoftwareCompliance`. Just create a file `x/compliance/types/device_software_compliance.go` in addition to `x/compliance/types/device_software_compliance.pb.go` . See for example `x/compliance/types/compliance_info.go`.